server/src/internalClusterTest/java/org/elasticsearch/indices/recovery/SnapshotBasedIndexRecoveryIT.java

```diff
@@ -496,6 +490,12 @@ public void testRecoveryIsCancelledAfterDeletingTheIndex() throws Exception {
                 }
             );
 
+            assertAcked(
```

Comment:
Is the difference between these two snippets is that we have to register a send behaviour before updating indices settings?

Reply:
yes, if we update settings first, we risk the recovery starting or completing before we add the send behavior. That makes the `recoverSnapshotFileRequestReceived.await()` a few lines down never complete, failing the test.

Reply:
We had an inadvertent race between the call to `addRequestHandlingBehavior` and the end of the recovery triggered by the update settings call - if the recovery completed before we got around to adding the request handler then we'd never count down the `recoverSnapshotFileRequestReceived` latch. The fix is to hook into the transport service before we even start the recovery.